kmip/tests/unit/services/server/test_engine.py

```diff
@@ -4435,6 +4435,55 @@ def test_destroy(self):
 
         e._data_session.commit()
 
+    def test_destroy_compromised_state(self):
+        """
+        Test that compromised object can be destroyed properly (put into
+        Destroyed Compromised state)
+        """
+        e = engine.KmipEngine()
+        e._data_store = self.engine
+        e._data_store_session_factory = self.session_factory
+        e._data_session = e._data_store_session_factory()
+        e._logger = mock.MagicMock()
+
+        key = (b'\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
+               b'\x00\x00\x00\x00\x00')
+        algorithm = enums.CryptographicAlgorithm.AES
+        obj = pie_objects.SymmetricKey(algorithm, 128, key)
+        obj.state = enums.State.COMPROMISED
+
+        e._data_session.add(obj)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        object_id = str(obj.unique_identifier)
+
+        # Test by specifying the ID of the object to destroy.
+        payload = destroy.DestroyRequestPayload(
+            unique_identifier=attributes.UniqueIdentifier(object_id)
+        )
+
+        response_payload = e._process_destroy(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Destroy"
+        )
+        self.assertEqual(
+            str(object_id),
+            response_payload.unique_identifier.value
+        )
+
+        symmetric_key = e._data_session.query(
+            pie_objects.SymmetricKey
+        ).filter(
+            pie_objects.ManagedObject.unique_identifier == object_id
+        ).one()
+
+        self.assertEqual(enums.State.DESTROYED_COMPROMISED,
+                         symmetric_key.state)
+
```

Comment:
I see how this ties in to how delete handled `COMPROMISED` objects above. I think it's ok to collapse this down into just a delete call on a `COMPROMISED` object. That code branch in delete will still get covered. We can test this explicitly later when we add more granularity on when to delete objects.